### PR TITLE
Remove undocumented and useless `air_equivalent` parameter

### LINKED ISSUE #13722: `air_equivalent` is undocumented
While looking through the `builtin` sources, I discovered the air node has a special field `air_equivalent = true`.

I don't know why it's there and this field is entirely undocumented. What's more confusing, it does not seem to be used anywhere in the code, neither in builtin nor C++ nor Minetest Game.

What is the purpose of this? If there is any purpose, I request that it'll be mentioned in `lua_api.md`.

### PR DESCRIPTION
**Pull request by Zughy**
Fixes #13722 . The parameter doesn't appear anywhere in the code, it doesn't do anything

## To do
This PR is Ready for Review.

## How to test
--

### CODE REVIEW

**Review by srifqi — APPROVED** [review]

**Review by grorp — APPROVED** [review]

**Comment by erlehmann** [comment]
I suggest to not merge this PR without figuring out which games and mods rely on this being truthy.

See https://github.com/minetest/minetest/issues/13722#issuecomment-1667846289 for more info.

**Comment by erlehmann** [comment]
# DO NOT MERGE THIS!

See https://github.com/minetest/minetest/issues/13722#issuecomment-1714241309 ff for findings revelant to this PR.

TL;DR:

* This property is not useful for the engine anymore.
* The property seems to be useful for mods.
* No one found a clear replacement yet.
